### ISSUE TITLE
Second argument to insertBefore may be undefined

### ISSUE THREAD

**Issue by thojanssens** [solidjs/solid#1265]
### Describe the bug

I am trying to test a Solid component with Vitest, happy-dom, and solid-testing-library. The component uses Portal.

There is an error because Solid sends an undefined value as second argument to `insertBefore`:

> ❯TypeError appendNodes : Failed to execute 'insertBefore' on 'Node': 2 arguments required, but only 1 present./my-app/node_modules/solid-js/web/dist/dev.js:478:60
 ❯ insertExpression /my-app/node_modules/solid-js/web/dist/dev.js:434:7
 ❯ Object.fn /my-app/node_modules/solid-js/web/dist/dev.js:409:17
 ❯ runComputation /my-app/node_modules/solid-js/dist/dev.js:691:22
 ❯ updateComputation /my-app/node_modules/solid-js/dist/dev.js:676:3
 ❯ createRenderEffect /my-app/node_modules/solid-js/dist/dev.js:205:75
 ❯ insertExpression /my-app/node_modules/solid-js/web/dist/dev.js:406:5
 ❯ Object.fn /my-app/node_modules/solid-js/web/dist/dev.js:214:33
 ❯ runComputation /my-app/node_modules/solid-js/dist/dev.js:691:22
 ❯ updateComputation /my-app/node_modules/solid-js/dist/dev.js:676:3
 ❯ createRenderEffect /my-app/node_modules/solid-js/dist/dev.js:205:75
 ❯ insert /my-app/node_modules/solid-js/web/dist/dev.js:214:3
 ❯ /my-app/node_modules/solid-js/web/dist/dev.js:99:37
 ❯ updateFn /my-app/node_modules/solid-js/dist/dev.js:168:15
 ❯ runUpdates /my-app/node_modules/solid-js/dist/dev.js:783:17
 ❯ createRoot /my-app/node_modules/solid-js/dist/dev.js:176:12
 ❯ render /my-app/node_modules/solid-js/web/dist/dev.js:97:3
 ❯ Module.render /my-app/node_modules/solid-testing-library/dist/index.mjs:29:11

I use Solid 1.5.7

### Your Example Website or App

-

### Steps to Reproduce the Bug or Issue

-

### Expected behavior

-

### Platform

- OS: Windows

**Comment by ryansolid** [comment]
I see.. yeah that works in all browsers as far as I can tell. Will HappyDOM accept `null` as the second argument for `insertBefore`? There is one location where we call insertBefore it looks like without the 2nd argument and I can just force it to `null` pretty easily.

**Comment by thojanssens** [comment]
Yes it will work then. Just to confirm, in node modules, solid, I replaced this line:
```javascript
for (let i = 0, len = array.length; i < len; i++) parent.insertBefore(array[i], marker);
```
by
```javascript
for (let i = 0, len = array.length; i < len; i++) parent.insertBefore(array[i], marker || null);
```
And it works now, the marker variable was undefined in my case.

**Comment by thojanssens** [comment]
Note that the more I read about it, I can understand that the spec requires 2 arguments, with the second possibly `null`. So we would follow the spec.

**Comment by LiQuidProQuo** [comment]
Fix trail: https://github.com/ryansolid/dom-expressions/commit/112b9518d597abc1bc2591854bd36a014757d93e

**Comment by ryansolid** [comment]
Merged. Will be fixed in 1.6 released.
